### ISSUE TITLE
Use Monochromatic Ids instead of hash to produce network identifiers.

### ISSUE THREAD

**Issue by greebie** [archivesunleashed/aut#440]
**Is your feature request related to a problem? Please describe.**

Using MD5 or other hashes for ids in Graphml and Gexf network outputs can result in lost data due to overloading. With RDDs it was possible to use .zipWithIndex(), but this function is not available in Dataframe.

**Describe the solution you'd like**

It may be possible to produce unique identifiers using `monotonicallyIncreasingId()`

https://stackoverflow.com/a/36946952/3050664

**Describe alternatives you've considered**

Other solutions implement zipWithId() in rdd and transform to df, including a schema to increase efficiency.

**Comment by ianmilligan1** [comment]
>lost data due to overloading

Hi @greebie - could you just flesh that out a bit for me?

**Comment by greebie** [comment]
Re: Overloading

For very large datasets, it is possible for two different urls to convert to the same hash, which means two different nodes will end up having the same id, causing data loss and potentially false links in the network.

**Comment by greebie** [comment]
(Also related to #243)

**Comment by SinghGursimran** [comment]
https://github.com/archivesunleashed/aut/pull/397

In this PR, I have implemented the code for getting a distinct node ID for each node.  
Though monotonicallyIncreasingId() function does provide monotonically increasing Id's, they are of the order of 10^10 which makes it difficult to manually visualize the graph.

Instead, I have used the Spark window concept to generate a distinct Id based on the URL of the node. That is each distinct URL will have a distinct nodeID.
